living_documentation_generator/generator.py

```diff
@@ -290,61 +291,42 @@ def _generate_markdown_pages(self, issues: dict[str, ConsolidatedIssue]) -> None
-        index_org_level_template = None
-        index_repo_level_template = None
+        is_structured_output = ActionInputs.get_is_structured_output_enabled()
+        is_grouping_by_topics = ActionInputs.get_is_grouping_by_topics_enabled()
+        output_path = ActionInputs.get_output_directory()
 
         # Load the template files for generating the Markdown pages
-        try:
-            with open(LivingDocumentationGenerator.ISSUE_PAGE_TEMPLATE_FILE, "r", encoding="utf-8") as f:
-                issue_page_detail_template = f.read()
-        except IOError:
-            logger.error("Issue page template file was not successfully loaded.", exc_info=True)
-
-        try:
-            with open(LivingDocumentationGenerator.INDEX_NO_STRUCT_TEMPLATE_FILE, "r", encoding="utf-8") as f:
-                index_page_template = f.read()
-        except IOError:
-            logger.error("Index page template file was not successfully loaded.", exc_info=True)
-
-        try:
-            with open(LivingDocumentationGenerator.INDEX_ROOT_LEVEL_TEMPLATE_FILE, "r", encoding="utf-8") as f:
-                index_root_level_page = f.read()
-        except IOError:
-            logger.error(
-                "Structured index page template file for root level was not successfully loaded.", exc_info=True
-            )
-
-        try:
-            with open(LivingDocumentationGenerator.INDEX_ORG_LEVEL_TEMPLATE_FILE, "r", encoding="utf-8") as f:
-                index_org_level_template = f.read()
-        except IOError:
-            logger.error(
-                "Structured index page template file for organization level was not successfully loaded.", exc_info=True
-            )
-
-        try:
-            with open(LivingDocumentationGenerator.INDEX_REPO_LEVEL_TEMPLATE_FILE, "r", encoding="utf-8") as f:
-                index_repo_level_template = f.read()
-        except IOError:
-            logger.error(
-                "Structured index page template file for repository level was not successfully loaded.", exc_info=True
-            )
+        (
+            issue_page_detail_template,
+            index_page_template,
+            index_root_level_page,
+            index_org_level_template,
+            index_repo_page_template,
+            index_data_level_template,
+        ) = self._load_all_templates()
 
         # Generate a markdown page for every issue
         for consolidated_issue in issues.values():
             self._generate_md_issue_page(issue_page_detail_template, consolidated_issue)
         logger.info("Markdown page generation - generated `%s` issue pages.", len(issues))
 
-        # Generate an index page with a summary table about all issues
-        if ActionInputs.get_is_structured_output_enabled():
-            self._generate_structured_index_pages(index_repo_level_template, index_org_level_template, issues)
+        # Generate all structure of the index pages
+        if is_structured_output:
```

Comment:
This if-elif is strange as both can happen in same time.

Reply:
I am not able to provide a better proposal too.
Let's fix it in a dedicated issue.